### PR TITLE
Replace boost::function with std::function

### PR DESCRIPTION
**Pull request by zancas**
Fixes: #4323

This logic is cherry-picked from the relevant commit, as referenced here:

https://github.com/bitcoin/bitcoin/pull/10395

### CODE REVIEW

**Comment by zancas** [comment]
@oxarbitrage do you have an opinion about this?

**Comment by zancas** [comment]
As far as I can tell the word "`slot`" is unnecessary in [these](https://github.com/zcash/zcash/blob/7f02ade06b93560c787291369e31ce21cfce479a/src/rpc/server.h#L28) invocations.

I'm tempted to add a commit to this PR that removes them, but maybe they have an effect I don't understand.

From what I read of the `std::function` syntax it seems like they are the name of the function object?   But these function objects are used as callbacks and can be anonymous?  So maybe `slot` can be removed?

**Comment by zancas** [comment]
I experimented with removing it, and go this error:

`Error: Unable to start HTTP server.`

**Comment by oxarbitrage** [comment]
A little part of this was done in https://github.com/zcash/zcash/pull/3858

The bitcoin PR for reference is https://github.com/bitcoin/bitcoin/pull/10395

**Comment by zancas** [comment]
Ahh...   thanks for the reference.   I think I will copy-paste it into the first comment, since that will incorporate it into the merge commit.

**Comment by zancas** [comment]
OK @oxarbitrage what's the next step to landing this?  (I edited the first comment using your link the relevant bitcoin PR.)

File: src/httpserver.cpp [hunk 1]
```diff
@@ -518,7 +518,7 @@ static void httpevent_callback_fn(evutil_socket_t, short, void* data)
         delete self;
 }
 
-HTTPEvent::HTTPEvent(struct event_base* base, bool deleteWhenTriggered, const boost::function<void(void)>& handler):
+HTTPEvent::HTTPEvent(struct event_base* base, bool deleteWhenTriggered, const std::function<void(void)>& handler):
```
Comment (str4d) [inline src/httpserver.cpp]: These removals in `httpserver.[cpp,h]` were implemented upstream as part of bitcoin/bitcoin#8421. Fine to remove here, any conflict should be simple to resolve.

File: src/scheduler.h [hunk 2]
```diff
@@ -39,7 +38,7 @@ class CScheduler
     CScheduler();
     ~CScheduler();
 
-    typedef boost::function<void(void)> Function;
+    typedef std::function<void(void)> Function;
```
Comment (str4d) [inline src/scheduler.h]: This change was implemented upstream as part of bitcoin/bitcoin#9605. Fine to remove here, any conflict should be simple to resolve.

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
Looks like a transient failure.

@zkbot retry

**Pull request merged**